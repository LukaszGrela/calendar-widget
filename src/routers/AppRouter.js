import React from 'react';
import { BrowserRouter, Switch, Route, Redirect } from 'react-router-dom';
import Page404 from '../pages/Page404';
import Home from '../pages/Home';
import moment from 'moment';

const AppRouter = () => (
    <BrowserRouter>
        <div>
            <header><h1>Calendar Widget</h1></header>
            <Switch>
                <Route exact path="(/|/index.html)" component={() => {
                    const now = moment();
                    return <Redirect to={`/${now.year()}/${now.month()}/${now.date()}`} />
                }} />
                <Route path="/:year?/:month?/:date?" component={Home} />
                <Route component={Page404} />
            </Switch>
            <footer>
                <div className='credits'>
                    Arrow Up/Down by Design Effectz from the Noun Project
                </div>
            </footer>
        </div>
    </BrowserRouter>
);

export default AppRouter;