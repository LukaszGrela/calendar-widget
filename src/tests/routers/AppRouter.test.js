import React from 'react';
// import { Redirect } from 'react-router-dom';
import { shallow } from 'enzyme';
import AppRouter from '../../routers/AppRouter';

test('Should render correctly', () => {
    const wrapper = shallow(<AppRouter />);
    // expect(wrapper.containsMatchingElement(<Redirect to='/' />)).toEqual(true);
    expect(wrapper).toMatchSnapshot();
});