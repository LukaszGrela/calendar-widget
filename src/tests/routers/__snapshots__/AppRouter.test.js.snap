// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
<BrowserRouter>
  <div>
    <header>
      <h1>
        Calendar Widget
      </h1>
    </header>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/index.html"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/router-calendar/:year?/:month?/:date?"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/linked-calendars"
      />
      <Route
        component={[Function]}
      />
    </Switch>
    <footer>
      <div
        className="credits"
      >
        Arrow Up/Down by Design Effectz from the Noun Project
      </div>
    </footer>
  </div>
</BrowserRouter>
`;
